pixeldata/src/lib.rs

```diff
@@ -2268,17 +2272,26 @@ where
             }
             DicomValue::Primitive(p) => {
                 // Non-encoded, just return the pixel data for a single frame
-                let frame_size = ((bits_allocated + 7) / 8) as usize
-                    * samples_per_pixel as usize
-                    * rows as usize
-                    * cols as usize;
-                let frame_offset = frame_size * frame as usize;
+                let frame_bits = rows as usize * cols as usize;
```

Comment:
`frame_bits` is only truly the number of bits in a frame if _Bits Allocated_ is 1 and _Samples Per Pixel_ is 1, it breaks in most other scenarios.

Perhaps we can redo the calculation starting from the number of pixels, then samples, and finish it off with the total number of pixels according to bits allocated.

```rust
let frame_pixels = rows as usize * cols as usize;
let frame_samples = frame_pixels * samples_per_pixel as usize;
let frame_size = if bits_allocated == 1 {
    frame_samples / 8
} else {
    frame_samples * ((bits_allocated + 7) / 8) as usize;
};
```